src/interfaces/swig/GaussianProcess.i

```diff
@@ -6,102 +6,24 @@
 
 /* Remove C Prefix */
 %shared_ptr(shogun::MeanFunction)
-%shared_ptr(shogun::ZeroMean)
-%shared_ptr(shogun::ConstMean)
-
 %shared_ptr(shogun::Inference)
 SHARED_RANDOM_INTERFACE(shogun::Inference)
-%shared_ptr(shogun::ExactInferenceMethod)
-%shared_ptr(shogun::LaplaceInference)
-%shared_ptr(shogun::SparseInference)
-%shared_ptr(shogun::SingleSparseInference)
-%shared_ptr(shogun::SingleFITCInference)
-%shared_ptr(shogun::SingleLaplaceInferenceMethod)
-%shared_ptr(shogun::MultiLaplaceInferenceMethod)
-%shared_ptr(shogun::FITCInferenceMethod)
-%shared_ptr(shogun::SingleFITCLaplaceInferenceMethod)
-%shared_ptr(shogun::VarDTCInferenceMethod)
-%shared_ptr(shogun::EPInferenceMethod)
-
 %shared_ptr(shogun::LikelihoodModel)
 SHARED_RANDOM_INTERFACE(shogun::LikelihoodModel)
-%shared_ptr(shogun::ProbitLikelihood)
-%shared_ptr(shogun::LogitLikelihood)
-%shared_ptr(shogun::SoftMaxLikelihood)
-%shared_ptr(shogun::GaussianLikelihood)
-%shared_ptr(shogun::StudentsTLikelihood)
-
-%shared_ptr(shogun::VariationalLikelihood)
-%shared_ptr(shogun::VariationalGaussianLikelihood)
-%shared_ptr(shogun::NumericalVGLikelihood)
-%shared_ptr(shogun::DualVariationalGaussianLikelihood)
-%shared_ptr(shogun::LogitVGLikelihood)
-%shared_ptr(shogun::LogitVGPiecewiseBoundLikelihood)
-%shared_ptr(shogun::LogitDVGLikelihood)
-%shared_ptr(shogun::ProbitVGLikelihood)
-%shared_ptr(shogun::StudentsTVGLikelihood)
-
-%shared_ptr(shogun::KLInference)
-%shared_ptr(shogun::KLLowerTriangularInference)
-%shared_ptr(shogun::KLCovarianceInferenceMethod)
-%shared_ptr(shogun::KLDiagonalInferenceMethod)
-%shared_ptr(shogun::KLCholeskyInferenceMethod)
-%shared_ptr(shogun::KLDualInferenceMethod)
-
+%shared_ptr(shogun::GaussianProcess)
```

Comment:
maybe we could move this somewhere else and delete the file?

Reply:
Machine.i

Reply:
If possible we should move these factory classes to factory.i no?

Reply:
@LiuYuHui ?